config.go

```diff
@@ -21,28 +21,21 @@ import (
 )
 
 // ParseUserInfo parses a user:password string into *url.Userinfo.
-// Username and password cannot be empty.
-func ParseUserInfo(val string) (*url.Userinfo, error) {
-	if val == "" {
-		return nil, nil //nolint:nilnil // nil is a valid value for Userinfo in URL
+func ParseUserInfo(in string) (*url.Userinfo, error) {
+	if in == "" {
+		return nil, fmt.Errorf("expected username[:password]")
 	}
-
-	u, p, ok := strings.Cut(val, ":")
-	if !ok {
-		return nil, fmt.Errorf("expected username:password")
+	if strings.Index(in, ":") != strings.LastIndex(in, ":") {
+		return nil, fmt.Errorf("only one colon is allowed")
```

Comment:
I think that can be potentially limiting.
The old logic says that username cannot have a colon which is more flexible I think.

Reply:
OK, removed this check.